### PR TITLE
Add PrivacyInfo.xcprivacy

### PR DESCRIPTION
**Pull request by michaelversus**
**Goal of this PR**
Comply to new rules of Apple for privacy. More info [here](https://developer.apple.com/documentation/bundleresources/privacy_manifest_files)

**Comments**
I counldn't find any apis that are used and should be added inside the PrivacyInfo file

### CODE REVIEW

File: Sources/PrivacyInfo.xcprivacy [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+<plist version="1.0">
+<dict>
+    <key>NSPrivacyTracking</key>
+    <false/>
+    <key>NSPrivacyTrackingDomains</key>
+    <array/>
+    <key>NSPrivacyCollectedDataTypes</key>
+    <array/>
+    <key>NSPrivacyAccessedAPITypes</key>
+    <array/>
```
Comment (martinK-7) [inline Sources/PrivacyInfo.xcprivacy:12]: I think we should add `NSPrivacyAccessedAPICategoryFileTimestamp` as described [here](https://developer.apple.com/documentation/bundleresources/privacy_manifest_files/describing_use_of_required_reason_api#4278393).
I can see at least `FileAttributeKey.modificationDate` is used in `CommonFunctions`

**Review by martinK-7 — COMMENTED** [review]
Hey, what is the status here? The date when apple requires it is coming soon.
@tsolomko Will there be a new release for it?

**Comment by tsolomko** [comment]
Hello @martinK-7,

I actually tried this week to understand what is this PrivacyInfo thing, and I am left quite confused.

First of all, with regards to your comment about `FileAttributeKey.modificationDate` in `CommonFunctions`. This file is not the part of the SWCompression _framework_. It is only included in the swcomp command-line tool, which is just an (elaborate) example of the usage of SWCompression.

In general, SWCompression by design does not interact directly with the filesystem, it does not collect any data and obviously does not send it anywhere. So the original PR with the "empty" PrivacyInfo file is correct, which leads me to the main point of confusion: I don't understand the point of all of this. Is it just to satisfy Apple's bureaucracy?

In addition, I don't understand how this file is supposed to interact with various methods of distribution of SWCompression. For instance, on Apple's webpage about this it is stated (I think) that if you include third-party libraries using SPM into your app, they still have to have this file included. But I don't see, how will SPM know about it, since it isn't specified anywhere in the Package.swift.

**Comment by martinK-7** [comment]
> Hello @martinK-7,
> 
> I actually tried this week to understand what is this PrivacyInfo thing, and I am left quite confused.
> 
> First of all, with regards to your comment about `FileAttributeKey.modificationDate` in `CommonFunctions`. This file is not the part of the SWCompression _framework_. It is only included in the swcomp command-line tool, which is just an (elaborate) example of the usage of SWCompression.
> 
> In general, SWCompression by design does not interact directly with the filesystem, it does not collect any data and obviously does not send it anywhere. So the original PR with the "empty" PrivacyInfo file is correct, which leads me to the main point of confusion: I don't understand the point of all of this. Is it just to satisfy Apple's bureaucracy?
> 
> In addition, I don't understand how this file is supposed to interact with various methods of distribution of SWCompression. For instance, on Apple's webpage about this it is stated (I think) that if you include third-party libraries using SPM into your app, they still have to have this file included. But I don't see, how will SPM know about it, since it isn't specified anywhere in the Package.swift.

Thanks for the quick answer and explanation. I am also confused by this whole topic, it's not 100% clear to me how this all comes together, and there is no way to check if your app complies unless you put your update into AppStore review. 
But if there is nothing to be stated for SWCompression, then I think you don't need the manifest at all, as stated [here](https://developer.apple.com/documentation/bundleresources/privacy_manifest_files):

```
[...] Otherwise, include a privacy manifest file in your third-party SDK if it uses required reasons API, collects data about the person using apps that include the third-party SDK, enables the app to collect data about people using the app, or contacts tracking domains.
```

To answer your question about SPM: You would need to distribute it via `resources` in Package.swift. As for example in [Alamofire](https://github.com/Alamofire/Alamofire/blob/master/Package.swift#L40).

**Comment by tsolomko** [comment]
Added in eabfe947ea59e8d24c93e33e8cbb6d1d5010c9a2 and released as part of [4.8.6](https://github.com/tsolomko/SWCompression/releases/tag/4.8.6) update. Thank you for the discussions!

**Pull request closed without merging**